### PR TITLE
NDW-3

### PR DESCRIPTION
**Pull request by brettvniekerk**
https://dltx.atlassian.net/jira/software/projects/NDW/boards/76/backlog?selectedIssue=NDW-3

### CODE REVIEW

**Review by OptimusOpus — DISMISSED** [review]
LGTM

**Review by OptimusOpus — DISMISSED** [review]

File: src/components/Mesh.tsx [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
+import { MeshMember, MeshModal } from "../components";
 
 const Mesh: React.FC = () => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [currentMeshMember, setCurrentMeshMember] = useState<MetadataEntry>();
+
+  const toggleModal = (event: React.SyntheticEvent) => {
+    const member = METADATA.filter(
+      (data) => data.name === event.currentTarget.id
+    );
```
Comment (Preston-Harrison) [inline src/components/Mesh.tsx:12]: `member.find` would be better here

File: src/components/MeshModal.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+type Props = {
+  currentMeshMember: MetadataEntry | undefined;
+  isModalOpen: boolean;
+  setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const MeshModal: React.FC<Props> = (props) => {
+  const { currentMeshMember, isModalOpen, setIsModalOpen } = props;
+
```
Comment (Preston-Harrison) [inline src/components/MeshModal.tsx:17]: if no mesh member, return null. Avoids lots of null checks

**Review by Preston-Harrison — CHANGES_REQUESTED** [review]

File: src/components/Mesh.tsx (hunk 1 above)
Comment (brettvniekerk) [reply src/components/Mesh.tsx:12]: 👍

**Review by OptimusOpus — APPROVED** [review]

**Review by Preston-Harrison — APPROVED** [review]

**Pull request merged**